### ISSUE TITLE
[BUG] oas tool 3.0 fail to treat optional field in schema as option.  

### ISSUE THREAD

**Issue by cll023** [oas-tools/oas-tools#356]
## **Describe the bug**
When we have a schema that defines a field of object type as optional, and if in the request body we don't include that optional field, oas tool complains.
## **To Reproduce**
1.  the schema of object definition as the following:

```components:
  schemas:
    TestDocument:
      type: object
      required:
      - version
      - id
      properties:
        version:
          description: Schema version
          type: string
        id:
          description: Unique ID
          type: string
          example: 1234567
        owner:
          $ref: '#/components/schemas/Owner'
    Owner:
      type: object
      required:
        - id
      properties:
        id:
          type: string
          format: uri
        name:
          type: string
          minLength: 1
          maxLength: 256
```
3. The following is the configuration to initiate the api server

```const http = require('http');
const express = require("express");
const { initialize } = require('@oas-tools/core');


const serverPort = 8000;
const app = express();
app.use(express.json({limit: '50mb'}));
const config = {
    middleware: {
        validator: {
            strict: true,
            requestValidation: true,
            responseValidation: true,
        },
    }
}


initialize(app, config).then(() => {
    http.createServer(app).listen(serverPort, () => {
    console.log("\nApp running at http://localhost:" + serverPort);
    console.log("________________________________________________________________");
    if (config.middleware.swagger?.disable !== false) {
        console.log('API docs (Swagger UI) available on http://localhost:' + serverPort + '/docs');
        console.log("________________________________________________________________");
    }
    });
});
```
4.  Send the following in a request body
{
  "version": "v1",
  "id": "12"
}
## **Expected behavior**
This should pass the schema validation. 
Instead, the API sever return 404 with the following error
{
    "error": "RequestValidationError: Request body does not match the schema specified in the OAS Document:\n- Validation failed at #/properties/owner/required > must have required property 'id'"
}
## **Additional context**
this is for  oas-tool version "3.0.3

The problem is with parseBody function that it would set "owner" value  as {} .  it results in  the schema validation.  The "owner" should be omitted in this case.

**Comment by alesancor1** [comment]
I'll look at it. Thanks for providing the code for reproducing the bug, it will make things a lot easier. Could you please edit your comment and add ``` to insert the code in a code block so it is easier to copy? :)

**Comment by pebo** [comment]
I'm having the same issue with validation failing for non-required objects. Any chance the https://github.com/oas-tools/oas-tools/pull/361 could be merged?

**Comment by alesancor1** [comment]
I will prioritize this issue for the next 3.0.X version. Many bugs have been reported in the last few months so I'll probably release a patch soon.

**Comment by raoulkent** [comment]
We've already ported to oas3, and are affected by this issue. Would it be possible to make incremental release as each item is fixed? #361 would be very helpful!

**Comment by alesancor1** [comment]
Yes, some conflicts have emerged in #361, asap they are solved it will be merged. There are some other problems that need to be solved, but they should not take much longer :)

**Comment by raoulkent** [comment]
That's great to hear. Do you have any time estimate?

I am not so well versed in the source code for OASv3, but please do let me know if there is something easy to pick up. I'd be happy to try and help.

**Comment by alesancor1** [comment]
I don't want to give any estimate since I am trying to work with many things in parallell, so it depends on the workload I have each moment.

I'd recommend you to read the docs and take some time looking at the code if you can, I tried to make it as easy to read as possible. Then, check the issues and PR and see if you can pick up something there. Also, if you find something isn't clear in docs, feel free to post an issue in the docs repo :)

PS: I merged #361, I'll try to fix the other issues and release a new version soon
